modules/video/src/videoplayer.cpp

```diff
@@ -0,0 +1,949 @@
+        "Video",
+        "Video",
+        "This should point to the video that should be played. It can "
+        "be either a file path or a url to a youtube video."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo PlayInfo = {
+        "Play",
+        "Play",
+        "Play video"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo PauseInfo = {
+        "Pause",
+        "Pause",
+        "Pause video"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo GoToStartInfo = {
+        "GoToStart",
+        "Go To Start",
+        "Go to start in video"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo ResetInfo = {
+        "Reset",
+        "Reset",
+        "Reset video"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo AudioInfo = {
+        "PlayAudio",
+        "Play Audio",
+        "Play audio"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo StartTimeInfo = {
+        "StartTime",
+        "Start Time",
+        "The date and time that the video should start in the format "
+        "'YYYY MM DD hh:mm:ss'."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo EndTimeInfo = {
+        "EndTime",
+        "End Time",
+        "The date and time that the video should end in the format "
+        "'YYYY MM DD hh:mm:ss'."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo PlaybackModeInfo = {
+        "PlaybackMode",
+        "Playback Mode",
+        "Determines the way the video should be played. The start and end time of the "
+        "video can be set, or the video can be played as a loop in real time."
+    };
+
+    struct [[codegen::Dictionary(VideoPlayer)]] Parameters {
+        // [[codegen::verbatim(VideoInfo.description)]]
+        std::string video;
+
+        // [[codegen::verbatim(AudioInfo.description)]]
+        std::optional<bool> playAudio;
+
+        // [[codegen::verbatim(StartTimeInfo.description)]]
+        std::optional<std::string> startTime [[codegen::datetime()]];
+
+        // [[codegen::verbatim(EndTimeInfo.description)]]
+        std::optional<std::string> endTime [[codegen::datetime()]];
+
+        enum class PlaybackMode {
+            MapToSimulationTime = 0,
+            RealTimeLoop
+        };
+
+        // The mode of how the video should be played back.
+        // Default is video is played back according to the set start and end times.
+        std::optional<PlaybackMode> playbackMode;
+    };
+#include "videoplayer_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+bool checkMpvError(int status) {
+    if (status < 0) {
+        LERROR(fmt::format("Libmpv API error: {}", mpv_error_string(status)));
+        return false;
+    }
+    return true;
+}
+
+void* getOpenGLProcAddress(void*, const char* name) {
+    return reinterpret_cast<void*>(
+        global::windowDelegate->openGLProcedureAddress(name)
+    );
+}
```

Comment:
These functions can be defined in the anonymous namespace above which makes them less visible to the outside